lib/compat.php

```diff
@@ -463,7 +463,7 @@ function gutenberg_inject_default_block_context( $args ) {
 			}
 
 			// Inject the query context if not done by Core.
-			$needs_query = empty( $block_type->uses_context ) && in_array( 'query', $block_type->uses_context, true );
+			$needs_query = is_array( $block_type->uses_context ) && in_array( 'query', $block_type->uses_context, true );
```

Comment:
Yeah, Im not very familiar with the PR that introduced this.  So I cant say whether this was supposed to be `! empty` or `||` instead of `&&` or what.  But `is_array` seems reasonable, and if its not the intention I'm sure @youknowriad will have a better idea on how to update that tomorrow.

Reply:
There's also this PR and comment, which is for some very similar code - https://github.com/WordPress/gutenberg/pull/27007#discussion_r524354381